### PR TITLE
Refactor SparseSliceOp to use a functor

### PR DESCRIPTION
**Pull request by benbarsdell**
This is in preparation for adding a GPU implementation.
No functional change.
Also adds a few tests for empty input/output sparse tensors.

cc @nluehr

### CODE REVIEW

File: tensorflow/python/kernel_tests/sparse_slice_op_test.py [hunk 1]
```diff
@@ -244,6 +250,25 @@ def testSliceAllColumns(self):
       self.assertAllEqual(sparse_tensor5.values, [5, 25, 35])
       self.assertAllEqual(sparse_tensor5.dense_shape, [4, 1])
 
+  @test_util.run_deprecated_v1
```
Comment (allenlavoie) [inline tensorflow/python/kernel_tests/sparse_slice_op_test.py:253]: Is there anything 1.x-specific about this? If you just leave off the decorator and remove the `self.session` it'll probably run in 2.x, which would be nice for a new test.

**Review by allenlavoie — APPROVED** [review]
@miaout17 for the Lite Flex delegate BUILD issue (see previous comment); any suggestions?

File: tensorflow/python/kernel_tests/sparse_slice_op_test.py (hunk 1 above)
Comment (benbarsdell) [reply tensorflow/python/kernel_tests/sparse_slice_op_test.py:253]: I don't think it's specific to 1.x, I just copied what the existing tests do. Will it need something like `with test_util.use_gpu():`?

Comment (allenlavoie) [reply tensorflow/python/kernel_tests/sparse_slice_op_test.py:253]: The existing test has use_gpu=False, so the equivalent would be `with ops.device("CPU"):` or `if test_util.is_gpu_available(): self.skipTest("Not tested on GPUs")`, with the latter saving some C/I resources.

Or if you want to run the test on GPUs you can use `with test_util.use_gpu():`, yes.

**Comment by benbarsdell** [comment]
I've changed the new test to run under v2, and I also worked out the TFLite build issue.

**Review by allenlavoie — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Pull request merged**